### DISCUSSION TITLE
Request DI.FM to be added to music providers

### DISCUSSION THREAD

**Discussion by jvenuto80** [music-assistant/support#1695] [Feature requests and ideas]
I have tried using the Tunein workaround but i get errors in the logs saying chromecast is not compatible but it plays just fine in tunein.  Can we add DI.FM as a provider that allows paid users to sign in?

**Comment by OzGav** [comment]
What TuneIn workaround?

**Reply by jvenuto80** [reply]
Add the stream url in the Tunein app under custom URL and it shows up in MA.  The custom URL plays fine in the tunein app but when casting to chromecast is says incompatible source in the MA logs

**Reply by OzGav** [reply]
If you have the stream url you don’t need to add it via tunein you can add it directly to MA via the radio view menu in the top right corner

**Comment by sparkydave1981** [comment]
+1 for Digitally Imported as a direct music provider!

**Comment by OzGav** [comment]
DI.fm and associated stations has been added to the 2.7 beta

**Reply by jvenuto80** [reply]
Love it!  Working great so far.  I might make one suggestion, not to auto favorite every channel when the key is added.  I didn't even know DI.FM had all those channels available since i just use the electronic music channels.

**Reply by OzGav** [reply]
Have a look at the docs https://beta.music-assistant.io/music-providers/digitally-incorporated/

**Reply by jvenuto80** [reply]
ah, i see.  Any idea when this will move into the non-beta stream?  Also, thanks for doing this.

**Reply by OzGav** [reply]
Usually a few months.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
